src/main/java/com/faforever/client/mapstruct/LeaderboardMapper.java

```diff
@@ -41,4 +48,41 @@
   @Mapping(target = "deviation", source = "rating.deviation")
   @Mapping(target = "mean", source = "rating.mean")
   LeaderboardRatingBean map(LeaderboardStats dto);
+
+  LeagueBean map(League dto, @Context CycleAvoidingMappingContext context);
+
+  League map(LeagueBean bean, @Context CycleAvoidingMappingContext context);
+
+  @Mapping(target = "leaderboard", source = "leagueLeaderboard")
+  LeagueSeasonBean map(LeagueSeason dto, @Context CycleAvoidingMappingContext context);
+
+  @Mapping(target = "leagueLeaderboard", source = "leaderboard")
+  LeagueSeason map(LeagueSeasonBean bean, @Context CycleAvoidingMappingContext context);
+
+  @Mapping(target = "index", source = "divisionIndex")
+  DivisionBean map(LeagueSeasonDivision dto, @Context CycleAvoidingMappingContext context);
+
+  @Mapping(target = "divisionIndex", source = "index")
+  LeagueSeasonDivision map (DivisionBean bean, @Context CycleAvoidingMappingContext context);
+
+  @Mapping(target = "leagueSeasonId", expression = "java(Integer.parseInt(dto.getLeagueSeasonDivision().getLeagueSeason().getId()))")
```

Comment:
```suggestion
  @Mapping(target = "leagueSeasonId", source = "leagueSeasonDivision.leagueSeason.id")
```

Will automatically include null safety in the generated method and convert the types for you

Reply:
When refactoring the leaderboardService I noticed that I don't really need the leagueSeasonId in the subdivision. I can get it by using the division that is attached to the subdivision

Reply:
So this property will just be removed?

Reply:
yes, unless you say that you want a reference to the season directly in the subdivisionBean

Reply:
Nah just making sure I read it right